bin/ch-run.c

```diff
@@ -145,8 +145,12 @@ int main(int argc, char * argv[])
    }
 
    setup_namespaces(args.container_uid, args.container_gid);
-   enter_udss(args.newroot, args.writable, args.binds,
+
+   char *newroot = realpath(args.newroot, NULL); // resolve relative paths
```

Comment:
Style:

* I recommend moving `realpath()` up to line 138 at the assignment to `args.newroot`.
* Comment can be eliminated.